src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -559,13 +627,29 @@ private static Details getDetails(DetailAST ast) {
                 case TokenTypes.LITERAL_DO:
                     details = getDetailsForDoLoops(ast);
                     break;
+                case TokenTypes.LITERAL_SWITCH:
+                    details = getDetailsForSwitch(ast);
```

Comment:
How are you ensuring we are only checking switch statements and not switch expressions?

Reply:
`final DetailAST checkExpression = switchNode.findFirstToken(TokenTypes.SWITCH_RULE);`
This statment will ensure that it is only switch statements

Reply:
@Kevin222004 please explain how we can determine that any given LITERAL_SWITCH token is an expression without checking its parent(s). Switch statements can have switch labeled rules.

Reply:
Just a few possible parents include method calls, any assignment operator, literal yield, etc.

Reply:
@nrmancuso as per my knowledge this switch is also a switch expression.
```
    public void test3() {
        int num = 2;

        switch (num) {
            case 1 -> System.out.println("One");
            default -> System.out.println("Other");
    } int a; } // ok

```
 please guide me if I am wrong.
I have removed the support of this as well. and the parent of this token is not a TOKENTYPES.EXPR

Reply:
Your example is a switch statement. See https://openjdk.org/jeps/361 for a detailed explanation.

Reply:
Thanks a lot

Reply:
done only expression has removed

Reply:
If this is done, then please re-answer the question.
> How are you ensuring we are only checking switch statements and not switch expressions?

From my view, I only see we are setting `nextToken` to `null` but are still returning a `Details` instance. I would have expected returning `null` for the `Details` instance saying there is nothing to validate, or after looking at the code, setting `Details`' `rcurly` to `null` which causes the current code to do no validation.

Reply:
That is what i am trying to tell you at https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1127545991 
if we set next token to null that this test is failing.

but thanks for the suggestion I have set rightcurly to null, i have not think about this i am only trying to find the next token 
but know it is fine 
Done

Reply:
But you understand we don't need any of the tokens if its an expression? We are not to validate expressions, they are to be invisible to this check for now.

Reply:
Yes I can understand that this check is not validating expression 
my confusion  for expression and statement become the reason for lots of lack in my communication but https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1128150468 know it is clear